rviz_mesh_tools_plugins/include/rviz_mesh_tools_plugins/MeshDisplay.hpp

```diff
@@ -405,6 +406,7 @@ private Q_SLOTS:
 
   /// Property to handle topic for meshMsg
   rviz_common::properties::RosTopicProperty* m_meshTopic;
+  rviz_common::properties::QosProfileProperty* m_meshTopicQos;
```

Comment:
Can you add a comment that this property is also used to set the QoS for the costs and colors as well?
Or maybe add properties and callbacks for setting QoS for costs and colors individually.

I think having QoS settings for each topic would be clearer for users. But it's more code and at least in our use-case, we don't need different QoS for costs, colors and mesh geometry.

But I think we should at least have a comment so we think of this behavior when we refactor things.

Reply:
resolved due to each topic having their own qos settings now